gce/storagemanager/gce.go

```diff
@@ -16,16 +17,29 @@ type gceStorageManager struct {
 }
 
 const (
-	// DriveTypeStandard is a constant for standard drive types
-	DriveTypeStandard = "pd-standard"
-	// DriveTypeSSD is a constant for ssd drive types
-	DriveTypeSSD = "pd-ssd"
-	// StandardIopsMultiplier is the amount with which a given pd-standard GiB size is multiplied
+	// GceDriveTypeStandard is a constant for standard drive types
+	GceDriveTypeStandard = "pd-standard"
+	// GceDriveTypeBalanced is a constant for pd-balanced drive types in GCE
+	GceDriveTypeBalanced = "pd-balanced"
+	// GceDriveTypeSSD is a constant for ssd drive types
+	GceDriveTypeSSD = "pd-ssd"
+	// GceStandardIopsMultiplier is the amount with which a given pd-standard GiB size is multiplied
 	// in order to get that drive's baseline IOPS performance
-	StandardIopsMultiplier = 0.75
-	// SSDIopsMultiplier is the amount with which a given ssd GiB size is multiplied
+	GceStandardIopsMultiplier = 0.75
+	// GceBalancedIopsMultiplier is the amount by which a given balanced disk's
+	// size is multiplied to get it's maximum IOPS performance.
+	GceBalancedIopsMultiplier = 6
+	// GceSSDIopsMultiplier is the amount with which a given ssd GiB size is multiplied
 	// in order to get that drive's baseline IOPS performance
-	SSDIopsMultiplier = 30
+	GceSSDIopsMultiplier = 30
+	// GceStandardMaxIops is the minimum of all the maximum iops that can be achieved with disk type px-standard.
+	// https://cloud.google.com/compute/docs/disks#:~:text=Standard%20persistent%20disks%20(%20pd%2Dstandard,that%20balance%20performance%20and%20cost.
+	GceStandardMaxIops uint64 = 7500
+	// GceBalancedMaxIops is the minimum of all the maximum iops that can be achieved with disk type px-balanced.
+	GceBalancedMaxIops uint64 = 30000
+	// GceSSDMaxIops is the minimum of all the maximum iops that can be achieved with disk type px-ssd
+	GceSSDMaxIops uint64 = 30000
```

Comment:
In the spec generator code below and  in the google docs, the Max SSD Iops is 15000.

Reply:
<img width="159" alt="image" src="https://user-images.githubusercontent.com/83608369/161609670-6a737db2-b4bc-456f-b23b-b1cea64cc61a.png">
The max iops is not a single number and it's a range. It depends on the vCPUs and IOsize. There is no direct correlation data that was available. So, I just went with the min of the max iops. Since we are not slicing the 15k-30k range, I thought I'll take the min of of the max iops.
Let me know if you prefer a different approach for this maxIOPS with a range. I thought of filing a JIRA which we can take a look at it later to include vCPUs and ioSizes into our equation. io-sizes might be challenging.

Reply:
May be max(max zonal read, max regional read, max zonal write, max regional write) is better. Will ping you on slack to discuss further.

Reply:
I have added *MaxIopsLeast and *MaxIopsMost. See if this implementation makes sense.